lightning-persister/src/util.rs

```diff
@@ -32,14 +57,32 @@ pub(crate) fn write_to_file<D: DiskWriteable>(path: String, filename: String, da
 		data.write_to_file(&mut f)?;
 		f.sync_all()?;
 	}
-	fs::rename(&tmp_filename, &filename_with_path)?;
 	// Fsync the parent directory on Unix.
 	#[cfg(not(target_os = "windows"))]
 	{
+		fs::rename(&tmp_filename, &filename_with_path)?;
 		let path = Path::new(&filename_with_path).parent().unwrap();
 		let dir_file = fs::OpenOptions::new().read(true).open(path)?;
 		unsafe { libc::fsync(dir_file.as_raw_fd()); }
 	}
+	#[cfg(target_os = "windows")]
+	{
+		let src = PathBuf::from(tmp_filename.clone());
+		let dst = PathBuf::from(filename_with_path.clone());
+		if Path::new(&filename_with_path.clone()).exists() {
+			unsafe {winapi::um::winbase::ReplaceFileW(
```

Comment:
Does `ReplaceFileW` work even if the destination file doesn't exist? If so, maybe its worth asking if rust upstream should be using that instead of MoveFileExW?

Reply:
> Does ReplaceFileW work even if the destination file doesn't exist?

I don't think so, or at least I get `background_processor` that never terminate: https://github.com/valentinewallace/rust-lightning/runs/1931376827